bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/VisitDocumentController.java

```diff
@@ -66,13 +73,27 @@ public VisitDocumentResponse save(@RequestBody VisitDocumentRequest visitDocumen
 
     @RequestMapping(method = RequestMethod.DELETE, value = baseVisitDocumentUrl)
     @ResponseBody
-    public void deleteDocument(@RequestParam(value = "filename") String fileName) {
-        if (Context.getUserContext().isAuthenticated()) {
-            if (StringUtils.isNotEmpty(fileName)) {
+    public ResponseEntity<Object> deleteDocument(@RequestParam(value = "filename") String fileName) {
+        if (Context.getUserContext().hasPrivilege(DELETE_PATIENT_DOCUMENT_PRIVILEGE)) {
```

Comment:
[Minor] I usually prefer to rewrite this as a guard clause which pulls all validations at start of the method body. Makes for lesser brackets/easier to read code. For instance: 

if (!hasPrivilege(DELETE_PATIENT_DOCUMENT)) {
    log. error();
    return BAD;
}

//Now there is  no need for else. The remaining code can be written as is.
XXX

Reply:
sure.

Reply:
Thanks!